### ISSUE TITLE
Can't dump SVD++ model

### ISSUE THREAD

**Issue by mkmacphe** [NicolasHug/Surprise#433]
#### Description
I'm trying to dump my trained SVD++ model using dump.dump, but I keep getting the error message
`TypeError: no default __reduce__ due to non-trivial __cinit__`

#### Steps/Code to Reproduce
```from surprise import Dataset
from surprise.model_selection import train_test_split
from surprise import SVDpp, dump

data = Dataset.load_builtin('ml-100k')
algo = SVDpp()

trainset, testset = train_test_split(data, test_size=0.2, random_state=42)

algo.fit(trainset)

dump.dump("test.surp", predictions=None, algo=algo, verbose=0)
```

#### Versions
Linux-5.15.0-46-generic-x86_64-with-glibc2.29
Python 3.8.10 (default, Mar 15 2022, 12:22:08) 
[GCC 9.4.0]
surprise 1.1.2

**Comment by ssifood** [comment]
we met the same issue, plz sovle this problem

**Comment by AnodeGrindYo** [comment]
I met the same issue with NMF too

**Comment by NicolasHug** [comment]
Thanks for the reports, I'll look into this. Meanwhile your best bet is to downgrade to v1.1.1

**Comment by AnodeGrindYo** [comment]
> Thanks for the reports, I'll look into this. Meanwhile your best bet is to downgrade to v1.1.1

Thank you, it works perfectly with v1.1.1

**Comment by Techie5879** [comment]
> Thanks for the reports, I'll look into this. Meanwhile your best bet is to downgrade to v1.1.1

Thanks, reverting to v1.1.1 works fine. Was pretty stumped about the problem

**Comment by NicolasHug** [comment]
Hi all, I just released version 1.1.3 where this issue should be fixed. Thank you for the reports and please let me know if it still breaks.
